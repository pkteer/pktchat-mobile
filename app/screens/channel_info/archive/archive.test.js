// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import Preferences from '@mm-redux/constants/preferences';
import {shallowWithIntl} from '@test/intl-test-helper';

import Archive from './archive';

jest.mock('@utils/theme', () => {
    const original = jest.requireActual('../../../utils/theme');
    return {
        ...original,
        changeOpacity: jest.fn(),
    };
});

describe('ChannelInfo -> Archive', () => {
    const baseProps = {
        canArchive: true,
        canUnarchive: false,
        channelId: '123',
        close: jest.fn(),
        deleteChannel: jest.fn(),
        displayName: 'Test Channel',
        getChannel: jest.fn(),
        handleSelectChannel: jest.fn(),
        isPublic: true,
        unarchiveChannel: jest.fn(),
        selectPenultimateChannel: jest.fn(),
        teamId: 'team-123',
        theme: Preferences.THEMES.denim,
        viewArchivedChannels: true,
    };

    test('should match snapshot for Archive Channel', () => {
        const wrapper = shallowWithIntl(
            <Archive
                {...baseProps}
            />,
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for Unarchive Channel', () => {
        const wrapper = shallowWithIntl(
            <Archive
                {...baseProps}
                canUnarchive={true}
            />,
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot Not render Archive', () => {
        const wrapper = shallowWithIntl(
            <Archive
                {...baseProps}
                canArchive={false}
                canUnarchive={false}
            />,
        );
        expect(wrapper.getElement()).toBeNull();
    });
});
