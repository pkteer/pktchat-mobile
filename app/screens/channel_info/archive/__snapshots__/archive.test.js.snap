// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelInfo -> Archive should match snapshot for Archive Channel 1`] = `
<React.Fragment>
  <Separator
    theme={
      Object {
        "awayIndicator": "#ffbc1f",
        "buttonBg": "#1c58d9",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3f4350",
        "codeTheme": "github",
        "dndIndicator": "#d24b4e",
        "errorTextColor": "#d24b4e",
        "linkColor": "#386fe5",
        "mentionBg": "#ffffff",
        "mentionColor": "#1e325c",
        "mentionHighlightBg": "#ffd470",
        "mentionHighlightLink": "#1b1d22",
        "newMessageSeparator": "#cc8f00",
        "onlineIndicator": "#3db887",
        "sidebarBg": "#1e325c",
        "sidebarHeaderBg": "#192a4d",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarTeamBarBg": "#14213e",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#5d89ea",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#28427b",
        "sidebarUnreadText": "#ffffff",
        "type": "Denim",
      }
    }
  />
  <channelInfoRow
    action={[Function]}
    defaultMessage="Archive Channel"
    icon="archive-outline"
    iconColor="#CA3B27"
    rightArrow={false}
    shouldRender={true}
    textColor="#CA3B27"
    textId="mobile.routes.channelInfo.delete_channel"
    theme={
      Object {
        "awayIndicator": "#ffbc1f",
        "buttonBg": "#1c58d9",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3f4350",
        "codeTheme": "github",
        "dndIndicator": "#d24b4e",
        "errorTextColor": "#d24b4e",
        "linkColor": "#386fe5",
        "mentionBg": "#ffffff",
        "mentionColor": "#1e325c",
        "mentionHighlightBg": "#ffd470",
        "mentionHighlightLink": "#1b1d22",
        "newMessageSeparator": "#cc8f00",
        "onlineIndicator": "#3db887",
        "sidebarBg": "#1e325c",
        "sidebarHeaderBg": "#192a4d",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarTeamBarBg": "#14213e",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#5d89ea",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#28427b",
        "sidebarUnreadText": "#ffffff",
        "type": "Denim",
      }
    }
    togglable={false}
  />
</React.Fragment>
`;

exports[`ChannelInfo -> Archive should match snapshot for Unarchive Channel 1`] = `
<React.Fragment>
  <Separator
    theme={
      Object {
        "awayIndicator": "#ffbc1f",
        "buttonBg": "#1c58d9",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3f4350",
        "codeTheme": "github",
        "dndIndicator": "#d24b4e",
        "errorTextColor": "#d24b4e",
        "linkColor": "#386fe5",
        "mentionBg": "#ffffff",
        "mentionColor": "#1e325c",
        "mentionHighlightBg": "#ffd470",
        "mentionHighlightLink": "#1b1d22",
        "newMessageSeparator": "#cc8f00",
        "onlineIndicator": "#3db887",
        "sidebarBg": "#1e325c",
        "sidebarHeaderBg": "#192a4d",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarTeamBarBg": "#14213e",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#5d89ea",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#28427b",
        "sidebarUnreadText": "#ffffff",
        "type": "Denim",
      }
    }
  />
  <channelInfoRow
    action={[Function]}
    defaultMessage="Unarchive Channel"
    icon="archive-arrow-up-outline"
    rightArrow={false}
    shouldRender={true}
    textId="mobile.routes.channelInfo.unarchive_channel"
    theme={
      Object {
        "awayIndicator": "#ffbc1f",
        "buttonBg": "#1c58d9",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3f4350",
        "codeTheme": "github",
        "dndIndicator": "#d24b4e",
        "errorTextColor": "#d24b4e",
        "linkColor": "#386fe5",
        "mentionBg": "#ffffff",
        "mentionColor": "#1e325c",
        "mentionHighlightBg": "#ffd470",
        "mentionHighlightLink": "#1b1d22",
        "newMessageSeparator": "#cc8f00",
        "onlineIndicator": "#3db887",
        "sidebarBg": "#1e325c",
        "sidebarHeaderBg": "#192a4d",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarTeamBarBg": "#14213e",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#5d89ea",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#28427b",
        "sidebarUnreadText": "#ffffff",
        "type": "Denim",
      }
    }
    togglable={false}
  />
</React.Fragment>
`;
